stdlib/gui/gui.lua

```diff
@@ -45,12 +47,22 @@ function Gui.Event.dispatch(event)
 
     local gui_element = event.element
     if gui_element and gui_element.valid then
+        local gui_element_name = gui_element.name;
+        local gui_element_state = nil;
+        local gui_element_text = nil;
+
+        if event.name == defines.events.on_gui_checked_state_changed then
+            gui_element_state = gui_element.state
+        end
+
+        if event.name == defines.events.on_gui_text_changed then
+            gui_element_text = gui_element.text
+        end
+
         for gui_element_pattern, handler in pairs(Gui.Event._registry[event.name]) do
-            local match_str = string.match(gui_element.name, gui_element_pattern)
```

Comment:
another option is have an optional parameter when you register the gui handler, whether you should receive events for 'invalid objects'